### ISSUE TITLE
generator.generate(model) Error:The first layer of your model must be the input layer

### ISSUE THREAD

**Issue by Mohsannaeem** [canyalniz/CNN-Inference-Didactic#2]
Hi, I am trying to generate inference model for CIfar10 dataset.Training code used from https://appliedmachinelearning.blog/2018/03/24/achieving-90-accuracy-in-object-recognition-task-on-cifar-10-dataset-with-keras-convolutional-neural-networks/
I had trained model and save sequential model using model.savemodel(model.h5)
As i Had no experience of CNN or machine learning.
 My question is  Whats the exactly mean of "The first layer of your model should be input layer??
Sorry for stupid question.....

**Comment by canyalniz** [comment]
Hello @Mohsannaeem ,

The truth is what you have experienced was entirely my fault. I made a mistake while preparing the interface. I have fixed the issue but I haven't had the time to test it thoroughly yet. You can start playing around with it right away or wait for a few days until I test it better.

One tip though, the model you are testing doesn't specify if the layers are *channels-first* or *channels-last*, this library only supports *channels-first*. If you want more details on this I would suggest you to look [here](https://stackoverflow.com/questions/46965047/channels-first-with-keras). Or you can simply insert this line right after the imports in the Python script: `keras.backend.set_image_data_format('channels_first')`

Also what you pointed out lead me to discover the need for even further improvements in my code. So thank you very much! Please feel free to ask more questions, your input is very valuable to me.

**Comment by Mohsannaeem** [comment]
Thanks for Your reply .I already changed my model and feed the input layer by modfying my code 
input_1 = Input(shape=input_shape) # input shape is 24,24,3 - needs to be 3,24,24


conv_1 = Conv2D(32, kernel_size=(3, 3),
             activation='relu',
             input_shape=input_shape)(input_1)

conv_2 = Conv2D(64, (3, 3), activation='relu')(conv_1)
............
